py-polars/src/conversion/any_value.rs

```diff
@@ -171,8 +172,8 @@ pub(crate) fn py_object_to_any_value<'py>(
     }
 
     fn get_bytes<'py>(ob: &Bound<'py, PyAny>, _strict: bool) -> PyResult<AnyValue<'py>> {
-        let value = ob.extract::<&'py [u8]>().unwrap();
-        Ok(AnyValue::Binary(value))
+        let value = ob.extract::<PyBackedBytes>().unwrap();
```

Comment:
Unfortunately this might be a performance regression, but I couldn't see any way to make this compile otherwise. Even if there were an API for this (eventually https://docs.rs/pyo3/latest/pyo3/conversion/trait.FromPyObjectBound.html might be less of a "don't use this" API?), this will also require changes to the callers since the objects aren't always preserved so there's no owner...

Reply:
Let's go directly to `Vec<u8>` then. Hopefully that saves a double alloc. `Pybackedbytes` goes via `Arc<[u8]>`.